### DISCUSSION TITLE
Dark mode transition effects trigger on page load

### DISCUSSION THREAD

**Discussion by Apostolique** [tailwindlabs/tailwindcss#3479] [Help]
### Describe the problem:

It looks like transition effects can sometimes trigger on page load when viewing a site's dark mode. The transition goes from light mode to dark mode. I was able to reproduce this in both Firefox, Chrome and Opera. I was not able to reproduce this locally (could be linked when CSS loading is delayed?).

### Link to a minimal reproduction:

Repository: <https://github.com/Apostolique/TailwindDarkTransitionIssue>
Live preview: <https://apostolique.github.io/TailwindDarkTransitionIssue/>

Make sure your system is set to dark mode. You can click on the rectangle to go to the next page. The transition doesn't happen in every case but I can usually trigger it after clicking a few times, or after waiting on the page for a brief delay. You'll see the rectangle transition from black to white. You'll also see the text's background transition even though it doesn't have a background set (except Chrome).

Edit: Recording of what I see in the Firefox browser: <https://gfycat.com/healthynicedotterel>.

**Comment by Doesntmeananything** [comment]
Same issue here. I'm using Next.js and injecting a theme detector script in the `head` element, so the flashing isn't caused by client-side hydration. 

Here's a clip where reloading the page with cache disabled triggers transition:


https://user-images.githubusercontent.com/34657799/103826627-3a5ed580-5088-11eb-92d0-f578c22b8a68.mp4

In my particular case, I'm also preloading fonts with `<link rel="preload" />`, which triggers flashing constantly. Without preloading, it happens very rarely, so it seems like some sort of race condition in the render phase.

Weirdly enough, adding a random script to the bottom of the `body` element kind of helps. I guess that delays whatever it is that triggers the transition just enough to prevent flashing most of the time.

**Reply by trm217** [reply]
I had the same problem but it was not Tailwinds fault. This problem mostly occures when programatically setting the dark value. The thing is, this only happens after,for example, react is initialized. This is the most common cause for this.

This package [next-themes](https://www.npmjs.com/package/next-themes) works perfectly and should solve your issue.

**Reply by Apostolique** [reply]
In my case I'm not running any JavaScript on the site. It's pure HTML and CSS.

**Reply by trm217** [reply]
The example you provided works fine for me, with no flashing whatsoever.

**Reply by Apostolique** [reply]
Which browser did you try it on? I think it wasn't as bad in Chrome.

**Reply by trm217** [reply]
Chrome & Firefox on desktop & Safari on iOS

**Reply by Apostolique** [reply]
This is what I see on Firefox on desktop: <https://gfycat.com/healthynicedotterel>.

**Comment by adamwathan** [comment]
@Apostolique my guess is that this is because you are shipping a 6mb CSS file and it takes the browser a long time to parse the whole thing and realize "oh crap dark mode is enabled". Try purging your CSS before shipping to production and seeing if it helps.

**Reply by Apostolique** [reply]
On my production site, purge is enabled.

**Comment by vietanhbui** [comment]
+1
